docs/datastorage/capabilities.md

```diff
@@ -43,27 +43,15 @@ There's two ways to obtain such an instance, through the Capability itself, or b
 
 The second method can be used to provide custom implementations. In the case of `IItemHandler`, the default implementation uses the `ItemStackHandler` class, which has an optional argument in the constructor, to specify a number of slots. However, relying on the existence of these default implementations should be avoided, as the purpose of the capability system is to prevent loading errors in contexts where the capability is not present, so instantiation should be protected behind a check testing if the capability has been registered (see the remarks about `@CapabilityInject` in the previous section).
 
-Once you have your own instance of the capability interface, you will want to notify users of the capability system that you expose this capability. This is done by overriding the `hasCapability` method, and comparing the instance with the capability you are exposing. If your machine has different slots based on which side is being queried, you can test this with the `facing` parameter. For Entities and ItemStacks, this parameter can be ignored, but it is still possible to have side as a context, such as different armor slots on a player (top side => head slot?), or about the surrounding blocks in the inventory (west => slot on the left?). Don't forget to fall back to `super`, otherwise the attached capabilities will stop working.
+Once you have your own instance of the capability interface, you will want to notify users of the capability system that you expose this capability and provide an optional of the interface reference. This is done by overriding the `getCapability` method, and comparing the instance with the capability you are exposing. If your machine has different slots based on which side is being queried, you can test this with the `side` parameter. For Entities and ItemStacks, this parameter can be ignored, but it is still possible to have side as a context, such as different armor slots on a player (top side => head slot?), or about the surrounding blocks in the inventory (west => slot on the left?). Don't forget to fall back to `super`, otherwise the attached capabilities will stop working.
 
 ```Java
 @Override
-public boolean hasCapability(Capability<?> capability, EnumFacing facing) {
-  if (capability == CapabilityItemHandler.ITEM_HANDLER_CAPABILITY) {
-    return true;
+public <T> LazyOptional<T> getCapability(Capability<T> cap, Direction side) {
+  if (cap == CapabilityItemHandler.ITEM_HANDLER_CAPABILITY) {
+    return (T) inventoryHandler.cast();
```

Comment:
```suggestion
    return LazyOptional.of(inventoryHandler).cast();
```
inventoryHandler needs to be passed to LazyOptional.of() and then cast.

Reply:
DO NOT Return a new LazyOptional every time -.-